### DISCUSSION TITLE
Rust plugin available

### DISCUSSION THREAD

**Discussion by BirgitPohl** [github/codeql#8485] [General]
I was wondering, since there is no native support for Rust, if the community can recommend a good Rust-plugin for LGTM.

**Comment by intrigus-lgtm** [comment]
As far as I know, there is no such thing as a Rust-plugin for LGTM or plugins for LGTM in general.
Can you clarify what you are interested in? Do you need a static analysis tool for Rust?

**Reply by BirgitPohl** [reply]
Exactly. LGTM covers almost all languages we are using, except for the one, we are using the most. Which is Rust.
And I'm currently looking for one code analysis that preferably rules them all to keep the learning curve low.

**Reply by hvitved** [reply]
Rust is not on the roadmap: https://github.com/orgs/github/projects/4247/views/1?filterQuery=label%3Acodeql.

**Reply by aibaars** [reply]
You might want to try [GitHub Code Scanning](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning) . Code Scanning offers an integrated user experience for many static analysis tools, including CodeQL, the analysis engine underlying LGTM. Using Code Scanning you can run the best tools for each language and view the results in a unified place.

**Reply by BirgitPohl** [reply]
I feel like, this is currently the best thing to do. Thanks!

**Reply by gogo2464** [reply]
there is no codeql rust plugin for lgtm because rust is not implemented yet for codeql... see https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/

**Comment by flipchan** [comment]
Is there a feature request to support Rust for codeql? If not, one should be started!

**Comment by izzeem** [comment]
So there is this https://github.com/github/codeql/discussions/10489

Not sure if anyone has worked on this or is interested in trying

**Comment by coadaflorin** [comment]
Hey everyone, just wanted to let you know that we shipped support for Rust in CodeQL to [public preview](https://github.blog/changelog/2025-06-30-codeql-support-for-rust-now-in-public-preview/). Give it a go and [let us know](https://github.com/orgs/community/discussions/161754#discussioncomment-13627726) how things went for you!

**Reply by flipchan** [reply]
Awesome!!

**Reply by boozook** [reply]
Thanks! Already integrated, will see...

Note, there is to ways how to enable this:
- simple button in the preferences near by Dependabot
- via job in a workflow yaml

Question:
Does it work with nightly toolchain?

**Reply by gogo2464** [reply]
Congrat! Let me see

**Reply by gogo2464** [reply]
For entertainement purpose solely.

**Reply by coadaflorin** [reply]
> Question:
> Does it work with nightly toolchain?

Are you referring to the Rust nightly toolchain or CodeQL?

**Reply by boozook** [reply]
> > Question:
> 
> > Does it work with nightly toolchain?
> 
> 
> 
> Are you referring to the Rust nightly toolchain or CodeQL? 

Rust nightly channel.

**Reply by coadaflorin** [reply]
It should work, but the latest features won't be supported and we don't test for it so things might fail once in a while.
